### PR TITLE
[eas-cli] display size of the archive

### PR DESCRIPTION
**Pull request by wkozyra95**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary.

# Why

https://linear.app/expo/issue/ENG-399/log-the-size-of-the-archived-project-when-uploading

# How

- add size to log after project archive is created
- add size to upload progress bar
- fix unhandled error in the tarball creation
- removed `--local` flag (it's unrelated to this PR, I just noticed that this flag is ignored either way)
- values larger than 100 are displayed as integer and smaller ones with one decimal point (e.g 700 MB , 55.3 MB) 

# Test Plan

2021-03-03-135217_575x89_scrot

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ This is the log of notable changes to EAS CLI and related packages.
 
 ### 🎉 New features
 
+- Display size of the upload when submitting build. ([#263](https://github.com/expo/eas-cli/pull/263) by [@wkozyra95](https://github.com/wkozyra95))
```
Comment (dsokal) [inline CHANGELOG.md:11]:
```suggestion
- Log the size of the archived project when uploading. ([#264](https://github.com/expo/eas-cli/pull/264) by [@wkozyra95](https://github.com/wkozyra95))
```

File: packages/eas-cli/src/build/utils/repository.ts [hunk 2]
```diff
@@ -119,7 +120,6 @@ async function makeProjectTarballAsync(): Promise<{ path: string; size: number }
   try {
     await spawnAsync('git', [
       'clone',
-      '--local',
```
Comment (dsokal) [inline packages/eas-cli/src/build/utils/repository.ts:122]: Why this change?

File: packages/eas-cli/src/utils/files.ts [hunk 3]
```diff
@@ -26,3 +26,32 @@ export async function maybeRenameExistingFileAsync(projectDir: string, filename:
     await fs.rename(desiredFilePath, path.resolve(projectDir, getRenamedFilename(filename, num)));
   }
 }
+
+export function formatBytes(bytes: number): string {
```
Comment (dsokal) [inline packages/eas-cli/src/utils/files.ts:30]: I'd prefer to use an npm package for that. E.g. https://github.com/optoolco/nice-bytes

File: packages/eas-cli/src/build/utils/repository.ts [hunk 4]
```diff
@@ -129,19 +129,26 @@ async function makeProjectTarballAsync(): Promise<{ path: string; size: number }
...
+    await fs.remove(shallowClonePath);
   }
+  clearTimeout(timer);
 
   const { size } = await fs.stat(tarPath);
+  const duration = endTimer(compressTimerLabel);
+  if (spinner.isSpinning) {
+    const prettyTime = formatMilliseconds(duration);
+    spinner.succeed(
+      `Compressed project files into ${formatBytes(size)} tar archive. ${chalk.dim(prettyTime)}`
```
Comment (dsokal) [inline packages/eas-cli/src/build/utils/repository.ts:148]:
```suggestion
      `Compressed project files ${chalk.dim(prettyTime)} (${formatBytes(size)})`
```
(According to https://linear.app/expo/issue/ENG-399/log-the-size-of-the-archived-project-when-uploading)

File: packages/eas-cli/src/build/utils/repository.ts (hunk 2 above)
Comment (wkozyra95) [reply packages/eas-cli/src/build/utils/repository.ts:122]: explained in PR description

File: packages/eas-cli/src/utils/files.ts (hunk 3 above)
Comment (wkozyra95) [reply packages/eas-cli/src/utils/files.ts:30]: discussed on slack

**Review by dsokal — APPROVED** [review]
